edb/edgeql/compiler/func.py

```diff
@@ -533,9 +533,18 @@ def compile_call_arg(arg_ql: qlast.Expr, *,
             dispatch.compile(arg_ql, ctx=fencectx),
             ctx=fencectx)
 
-        return setgen.scoped_set(
-            setgen.ensure_stmt(arg_ir, ctx=fencectx),
-            ctx=fencectx)
+        stmt = setgen.ensure_stmt(arg_ir, ctx=fencectx)
+        if (ctx.expr_exposed and ctx.implicit_limit
+                and isinstance(stmt, irast.SelectStmt) and not stmt.limit):
+            stmt.limit = setgen.ensure_set(
```

Comment:
It looks like `dispatch.compile` already calls `ensure_set` at least for constants.

Reply:
Mypy doesn’t understand singledispatch